src/ConnectedMode/UI/Credentials/CredentialsWnd.xaml

```diff
@@ -0,0 +1,158 @@
+                    <TextBlock VerticalAlignment="Center">
+                        <!-- NavigateUri has to be filled, for the RequestNavigate to be triggered. The real Uri is calculated in the ViewModel --> 
+                        <Hyperlink NavigateUri="https://sonarcloud.io/account/security" RequestNavigate="GenerateToken_Navigate">Generate</Hyperlink>
+                    </TextBlock>
+                    <imaging:CrispImage Width="16" Height="16" Margin="2,5,0,0" Cursor="Hand" Moniker="{x:Static vsimagecatalog:KnownMonikers.Link}"
+                                        MouseLeftButtonDown="GenerateLinkIcon_Click"/>
+                </StackPanel>
+            </Grid>
+
+
+            <TextBlock Grid.Row="1" Grid.Column="0" Text="{x:Static ui:UiResources.Username}" VerticalAlignment="Center" Margin="0,15,10,0"
+                       Visibility="{Binding  Path=IsCredentialsAuthentication, Converter={StaticResource TrueToVisibleConverter}}"/>
+            <Border  Grid.Row="1" Grid.Column="1" Margin="0,15,0,0">
+                <Border.Style>
+                    <Style TargetType="Border">
+                        <Style.Triggers>
+                            <DataTrigger Binding="{Binding ShouldUsernameBeFilled}" Value="True">
+                                <Setter Property="BorderBrush" Value="{StaticResource InvalidText}"></Setter>
+                                <Setter Property="BorderThickness" Value="1"></Setter>
+                            </DataTrigger>
+                        </Style.Triggers>
+                    </Style>
+                </Border.Style>
+                <TextBox Text="{Binding Path=Username, UpdateSourceTrigger=PropertyChanged}" Margin="0"
+                         Visibility="{Binding  Path=IsCredentialsAuthentication, Converter={StaticResource TrueToVisibleConverter}}"/>
+            </Border>
+
+            <TextBlock Grid.Row="2" Grid.Column="0" Text="{x:Static ui:UiResources.Password}" VerticalAlignment="Center" Margin="0,5,10,0"
+                       Visibility="{Binding  Path=IsCredentialsAuthentication, Converter={StaticResource TrueToVisibleConverter}}"/>
+            <Border Grid.Row="2" Grid.Column="1" Margin="0,5,0,0">
+                <Border.Style>
+                    <Style TargetType="Border">
+                        <Style.Triggers>
+                            <DataTrigger Binding="{Binding ShouldPasswordBeFilled}" Value="True">
+                                <Setter Property="BorderBrush" Value="{StaticResource InvalidText}"></Setter>
+                                <Setter Property="BorderThickness" Value="1"></Setter>
+                            </DataTrigger>
+                        </Style.Triggers>
+                    </Style>
+                </Border.Style>
+                <TextBox Text="{Binding Path=Password, UpdateSourceTrigger=PropertyChanged}" Margin="0" 
```

Comment:
Use PasswordBox

Reply:
https://github.com/SonarSource/sonarlint-visualstudio/blob/546208681c669cc11747194c4aa6e9d66c120dd9/src/Integration/Connection/UI/ConnectionInfoDialogView.xaml#L190

Reply:
Makes sense to use PasswordBox also for Token